packages/react-instantsearch-core/src/widgets/InstantSearch.tsx

```diff
@@ -246,7 +256,15 @@ class InstantSearch extends Component<Props, State> {
     }
   }
 
-  onSearchParameters(getSearchParameters, context, props) {
+  onSearchParameters(
+    getSearchParameters: ConnectorDescription['getMetadata'],
+    context: {
+      ais: InstantSearchContext;
+      multiIndexContext: IndexContext;
+    },
+    props: object,
+    getMetadata?: ConnectorDescription['getMetadata']
```

Comment:
I see `getMetadata` being optional here, but `widgetsCollector` is expecting it to be required.

Reply:
good call, it's optional in both places

Reply:
actually ConnectorDescription already makes it optional, so this was duplicate